### PR TITLE
LPS-183153 change to correct action on caseResults listview

### PR DESCRIPTION
**Pull request by welitonpe**

### CODE REVIEW

**Comment by welitonpe** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/CompareRuns/useCompareRuns.ts [hunk 1]
```diff
@@ -60,10 +60,10 @@ const useCompareRuns = (
 	const operator = type === 'details' ? '' : type;
 
 	const {data} = useFetch<CompareRunsResponse>(
-		`/o/c/compare-runs/${runA}/${runB}/${operator}`,
+		`/compare-runs/${runA}/${runB}/${operator}`,
 		{
 			params: {customParams: {componentId, teamId}},
-			swrConfig: {shouldFetch: false},
+			swrConfig: {shouldFetch: true},
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/CompareRuns/useCompareRuns.ts:66]: useFetch will always request data, unless shouldFetch is false

```suggestion
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/useCaseActions.ts [hunk 2]
```diff
@@ -43,11 +43,14 @@ const useCaseActions = ({isHeaderActions}: ActionsHookParameter = {}) => {
...
 		},
 		{
-			action: ({id}, mutate) =>
+			action: ({id}, mutate) => {
 				deleteResource(`/cases/${id}`)
 					?.then(() => removeItemFromList(mutate, id))
 					.then(form.onSuccess)
-					.catch(form.onError),
+					.catch(form.onError);
+				isHeaderActions && navigate('../');
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/useCaseActions.ts:51]: No conditional actions, use if instead

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Requirements/useRequirementActions.ts [hunk 3]
```diff
@@ -51,11 +51,13 @@ const useRequirementActions = ({
...
 		},
 		{
-			action: ({id}, mutate) =>
+			action: ({id}, mutate) => {
 				deleteResource(`/requirements/${id}`)
 					?.then(() => removeItemFromList(mutate, id))
 					.then(onSave)
-					.catch(onError),
+					.catch(onError);
+				isHeaderActions && navigate('../');
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Requirements/useRequirementActions.ts:59]: No conditional actions, use if instead

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildForm/index.tsx [hunk 4]
```diff
@@ -51,13 +51,11 @@ type OutletContext = {
...
-		buildTemplate,
-		buildTemplateId,
-		projectId,
-		routineId,
-	} = useParams();
+	const {buildId, buildTemplateId, projectId, routineId} = useParams();
+
+	const [searchParams] = useSearchParams();
+
+	const buildTemplate = searchParams.get(`template`);
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildForm/index.tsx:58]:
```suggestion
	const buildTemplate = searchParams.get('template');
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/Inner/CaseResult/useCaseResultActions.tsx [hunk 5]
```diff
@@ -28,13 +29,16 @@ const useCaseResultActions = (
...
 	const actionsRef = useRef([
 		{
-			action: ({id}, mutate) =>
+			action: ({id}, mutate) => {
 				deleteResource(`/caseresults/${id}`)
 					?.then(() => removeItemFromList(mutate, id))
 					.then(form.onSave)
-					.catch(form.onError),
+					.catch(form.onError);
+				isHeaderActions && navigate('../');
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/Inner/CaseResult/useCaseResultActions.tsx:40]: No conditional actions, use if instead

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/useBuildActions.ts [hunk 6]
```diff
@@ -81,11 +81,13 @@ const useBuildActions = ({isHeaderActions}: ActionsHookParameter = {}) => {
...
 		},
 		{
-			action: ({id}, mutate) =>
+			action: ({id}, mutate) => {
 				deleteResource(`/builds/${id}`)
 					?.then(() => removeItemFromList(mutate, id))
 					.then(modal.onSave)
-					.catch(modal.onError),
+					.catch(modal.onError);
+				isHeaderActions && navigate('../');
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/useBuildActions.ts:89]: No conditional actions, use if instead

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/useRoutineActions.tsx [hunk 7]
```diff
@@ -66,11 +66,13 @@ const useRoutineActions = ({isHeaderActions}: ActionsHookParameter = {}) => {
...
 		},
 		{
-			action: ({id}, mutate) =>
+			action: ({id}, mutate) => {
 				deleteResource(`/routines/${id}`)
 					?.then(() => removeItemFromList(mutate, id))
 					.then(form.onSuccess)
-					.catch(form.onError),
+					.catch(form.onError);
+				isHeaderActions && navigate('../');
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/useRoutineActions.tsx:74]: No conditional actions, use if instead

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Suites/useSuiteActions.ts [hunk 8]
```diff
@@ -37,11 +37,13 @@ const useSuiteActions = ({isHeaderActions}: ActionsHookParameter = {}) => {
...
 		{
-			action: ({id}, mutate) =>
+			action: ({id}, mutate) => {
 				deleteResource(`/suites/${id}`)
 					?.then(() => removeItemFromList(mutate, id))
 					.then(form.onSuccess)
-					.catch(form.onError),
+					.catch(form.onError);
+				isHeaderActions && navigate('../');
+			},
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Suites/useSuiteActions.ts:46]: No conditional actions, use if instead

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Review by kevenleone — APPROVED** [review]
Thank you @welitonpe 

I did minor changes to the code, see: https://github.com/pt-liferay-solutions/liferay-portal/pull/4065

**Comment by kevenleone** [comment]
resent: https://github.com/pt-liferay-solutions/liferay-portal/pull/4065

**Pull request closed without merging**